Sources/Tentacle/Client.swift

```diff
@@ -196,10 +196,14 @@ public final class Client {
             .flatMap(.concat) { data, response -> SignalProducer<(Response, Any), Error> in
                 let response = response as! HTTPURLResponse
                 let headers = response.allHeaderFields as! [String:String]
-                return SignalProducer
-                    .attempt {
-                        return JSONSerialization.deserializeJSON(data).mapError { Error.jsonDeserializationError($0.error) }
-                    }
+
+                // The explicitness is required to pick up
+                // `init(_ action: @escaping () -> Result<Value, Error>)`
+                // over `init(value: Value)`.
+                let producer: SignalProducer<Any, Error> = SignalProducer { () -> Result<Any, Error> in
+                    return JSONSerialization.deserializeJSON(data).mapError { Error.jsonDeserializationError($0.error) }
+                }
```

Comment:
```swift
let producer: SignalProducer<Any, Error> = SignalProducer {
    return JSONSerialization.deserializeJSON(data).mapError { Error.jsonDeserializationError($0.error) }
}
```

is not sufficient because `Result<Any, Error>` itself is treated as `Any`.